### PR TITLE
generate address qrcode

### PR DESCRIPTION
**Pull request by dangershony**

### CODE REVIEW

File: src/Angor/Client/Pages/Wallet.razor [hunk 1]
```diff
@@ -520,7 +523,30 @@
...
+        Task.Run(() =>
+        {
+            var address = localAccountInfo.GetNextReceiveAddress();
+
+            base64qrcode = Util.GenerateQRCode(address);
+            lastqrcode = base64qrcode;
+            lastaddress = address;
+
+            StateHasChanged();
+        });
```
Comment (DavidGershony) [inline src/Angor/Client/Pages/Wallet.razor:547]: This should really be in it's own component

File: src/Angor/Client/Shared/Util.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
+﻿using Angor.Client.Storage;
+using Angor.Shared.Models;
+using Microsoft.AspNetCore.Components;
+using QRCoder;
+
+namespace Angor.Client.Shared
+{
+    public class Util
+    {
+        public static string GenerateQRCode(string content)
```
Comment (DavidGershony) [inline src/Angor/Client/Shared/Util.cs:10]: Why is it static and not injected?

**Comment by dangershony** [comment]
I now consider this should be a popup

**Review by DavidGershony — APPROVED** [review]

File: src/Angor/Client/Pages/Wallet.razor [hunk 3]
```diff
@@ -416,7 +418,11 @@
             storage.SetAccountInfo(network.Name, accountInfo);
             
             localAccountInfo = accountInfo;
-            
+
+            nextReceiveAddress = localAccountInfo.GetNextReceiveAddress();
+
+            showQrCode.GenerateQRCode(nextReceiveAddress);
```
Comment (DavidGershony) [inline src/Angor/Client/Pages/Wallet.razor:424]: Should this be awaited?

File: src/Angor/Client/Components/ShowQrCode.razor [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    private static string lastqrcode;
+    private static string lastaddress;
+    private string base64qrcode;
+
+    protected override async Task OnInitializedAsync()
+    {
+        GenerateQRCode(Data);
+    }
+
+    public Task GenerateQRCode(string newData)
```
Comment (DavidGershony) [inline src/Angor/Client/Components/ShowQrCode.razor:30]: GenerateQRCodeAsync

File: src/Angor/Client/Pages/Wallet.razor (hunk 3 above)
Comment (dangershony) [reply src/Angor/Client/Pages/Wallet.razor:424]: no I don't want it to clock the caller

**Comment by dangershony** [comment]
@DavidGershony I think I will change this PR to use a popup instead so put it on hold

**Pull request merged**